agents/src/inference/llm.ts

```diff
@@ -13,124 +13,104 @@ import * as llm from '../llm/index.js';
+  | 'openai/gpt-4.1'
+  | 'openai/gpt-4.1-mini'
+  | 'openai/gpt-4.1-nano'
+  | 'openai/gpt-4o'
+  | 'openai/gpt-4o-mini'
+  | 'openai/gpt-oss-120b';
+
+export type GoogleModels = 'google/gemini-2.0-flash-lite';
+
+export type QwenModels = 'qwen/qwen3-235b-a22b-instruct';
+
+export type KimiModels = 'moonshotai/kimi-k2-instruct';
+
+export type DeepSeekModels = 'deepseek-ai/deepseek-v3';
+
+type ChatCompletionPredictionContentParam = OpenAI.Chat.Completions.ChatCompletionPredictionContent;
+type WebSearchOptions = OpenAI.Chat.Completions.ChatCompletionCreateParams.WebSearchOptions;
+type ToolChoice = OpenAI.Chat.Completions.ChatCompletionCreateParams['tool_choice'];
+type Verbosity = 'low' | 'medium' | 'high';
+
+export interface ChatCompletionOptions extends Record<string, unknown> {
+  frequency_penalty?: number;
+  logit_bias?: Record<string, number>;
+  logprobs?: boolean;
+  max_completion_tokens?: number;
+  max_tokens?: number;
+  metadata?: Record<string, string>;
+  modalities?: Array<'text' | 'audio'>;
+  n?: number;
+  parallel_tool_calls?: boolean;
+  prediction?: ChatCompletionPredictionContentParam | null;
+  presence_penalty?: number;
+  prompt_cache_key?: string;
+  reasoning_effort?: 'minimal' | 'low' | 'medium' | 'high';
+  safety_identifier?: string;
+  seed?: number;
+  service_tier?: 'auto' | 'default' | 'flex' | 'scale' | 'priority';
+  stop?: string | string[];
+  store?: boolean;
+  temperature?: number;
+  top_logprobs?: number;
   top_p?: number;
-}
+  user?: string;
+  verbosity?: Verbosity;
+  web_search_options?: WebSearchOptions;
 
-export interface BasetenOptions {
-  top_p?: number;
+  // livekit-typed arguments
+  tool_choice?: ToolChoice;
+  // TODO(brian): support response format
+  // response_format?: OpenAI.Chat.Completions.ChatCompletionCreateParams['response_format']
 }
 
-export type LLMModels = AzureModels | CerebrasModels | GroqModels | BasetenModels | AnyString;
-
-export type LLMOptions<T extends LLMModels> = T extends AzureModels
-  ? AzureOptions
-  : T extends CerebrasModels
-    ? CerebrasOptions
-    : T extends GroqOptions
-      ? GroqOptions
-      : T extends BasetenOptions
-        ? BasetenOptions
-        : Record<string, unknown>;
-
-export type Verbosity = 'low' | 'medium' | 'high';
-const DEFAULT_BASE_URL = 'https://agent-gateway.livekit.cloud/v1';
-
-export interface InferenceLLMOptions<TModel extends LLMModels> {
-  model: TModel;
-  temperature?: number;
-  parallelToolCalls?: boolean;
-  toolChoice?: llm.ToolChoice;
-  maxCompletionTokens?: number;
+export type LLMModels =
+  | OpenAIModels
+  | GoogleModels
+  | QwenModels
+  | KimiModels
+  | DeepSeekModels
+  | AnyString;
+
+export interface InferenceLLMOptions {
+  model: LLMModels;
+  provider?: string;
   baseURL: string;
   apiKey: string;
   apiSecret: string;
-  verbosity?: Verbosity;
-  extraKwargs: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  extraKwargs: ChatCompletionOptions;
```

Comment:
I think we should change this to whatever is native for JS? `extra_kwargs` is really a python term. is it `modelOptions` for python?